### PR TITLE
[WIP] doc: Add documentation for size optimized read balancer

### PR DESCRIPTION
**Pull request by ljflores**
Work in progress; adds documentation for the feature added in https://github.com/ceph/ceph/pull/55198.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by ljflores** [comment]
> LGTM

It's still a work in progress so I may re-request your review when it has more work.

File: doc/dev/balancer-design.rst [hunk 1]
```diff
@@ -46,12 +46,18 @@ to achieve overall cluster performance in heterogeneous systems.
...
-Plans for the Next Version
---------------------------
+The Size Optimized Read Balancer
+--------------------------------
+
+Available in Squid and beyond, the size optimized read balancer introduces an improvement to the read balancer which increases the cluster performance when
+all the devices in a pool have similar throughput (IOPs and bandwidth) but are not identical in their capacities.
 
-1. Improve behavior for heterogeneous OSDs in a pool
+This is a common use case for physical devices in which the throughput of devices with the same technology (e.g. HDD, SSD, NVMe SSD) is similar, unrelated
```
Comment (ljflores) [inline doc/dev/balancer-design.rst:61]: Add "for deeper details..." link to separate page.

File: doc/rados/operations/pools.rst [hunk 2]
```diff
@@ -574,6 +574,14 @@ You may set values for the following keys:
...
    :Default: ``0``
 
+.. _read_ratio:
+
+.. describe:: read_ratio
+
+   :Description: Sets the percent of read IOs out of all IOs for a pool. Applicable to replicated pools only. This value is used to improve read balancing when OSDs have different weights.
+
+   :Type: Integer
+   :Default: ``70``
```
Comment (ljflores) [inline doc/rados/operations/pools.rst:584]: Change this; default is 0 (unset)

Comment (zdover23) [reply doc/rados/operations/pools.rst]: @ljflores Shall I make these changes?

Comment (ljflores) [reply doc/rados/operations/pools.rst]: @zdover23 no need to make these changes. I'll get your final review when the doc updates are ready, but it needs work from me and Josh.

Comment (zdover23) [reply doc/rados/operations/pools.rst]: @ljflores, Roger that. I'll add this to my to-do list, two weeks from now.

**Review by JoshSalomon — APPROVED** [review]

**Pull request closed without merging**